### PR TITLE
[schema] FileSource: `privacy`, `id` + removal of `custom` field

### PR DESCRIPTION
**Pull request by Barackos**
## Purpose:
- FileSource's `privacy` field + docs
- Deprecation of `custom` field
- Creation of `id` and `json` fields

## Current flow ("ricos") vs. New flow ("new ricos"):
image (11)

### CODE REVIEW

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto [hunk 1]
```diff
@@ -6,9 +6,15 @@ import "wix/rich_content/v1/common.proto";
...
+    UNSET = 0;
+    PUBLIC = 1;
+    PRIVATE = 2;
+  }
   PluginContainerData container_data = 1;
   FileSource src = 2;
   google.protobuf.StringValue name = 3;
   google.protobuf.StringValue type = 4;
   google.protobuf.Int32Value size = 5;
+  Privacy privacy = 6;
```
Comment (mharsat) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto:19]: I think `privacy` should be part of the `FileSource` or `Media` types.

1. It's paired to the source as it describes how it should be resolved
2. Once we decide to support private media for more types of media we'd want a general enum for it (prevent future break)

@shaulgo WDYT?

File: packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts [hunk 2]
```diff
@@ -186,10 +187,16 @@ const convertMention = (data: Partial<MentionData> & { mention }) => {
   delete data.slug;
 };
 
-const convertFileData = (data: FileData & FileComponentData) => {
+const convertFileData = (data: Omit<FileData, 'privacy'> & FileComponentData) => {
+  const { privacy } = data;
   const { url, custom } = data.src || {};
   data.url = url;
   data.id = custom;
+  if (!privacy || (privacy as FileData_Privacy) === 'UNSET') {
```
Comment (mharsat) [inline packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts:195]:
```ts
privacy === FileData_Privacy.UNSET
```

File: packages/ricos-content/web/src/converters/draft/fromDraft/convertRicosPluginData.ts [hunk 3]
```diff
@@ -168,6 +169,7 @@ const convertMention = (data: {
 const convertFileData = (data: FileComponentData & { src }) => {
   const src: FileSource = { url: data.url, custom: data.id };
   data.src = src;
+  data.privacy = (data.privacy?.toUpperCase() || 'UNSET') as MediaPrivacy;
```
Comment (mharsat) [inline packages/ricos-content/web/src/converters/draft/fromDraft/convertRicosPluginData.ts:172]:
```ts
FileData_Privacy.UNSET
```

File: packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts (hunk 2 above)
Comment (Barackos) [reply packages/ricos-content/web/src/converters/draft/toDraft/convertDraftPluginData.ts:195]: TS error, but got the idea, will do it a little different:
```js
(privacy as string) === FileData_Privacy.UNSET
```

Screen Shot 2021-07-01 at 16 13 16

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto (hunk 1 above)
Comment (shaulgo) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/plugin_file.proto:19]: I agree. But I need to understand the use case better. Let's talk about it. Maybe custom src is already enough.

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 4]
```diff
@@ -73,10 +73,18 @@ message PluginContainerData {
...
 message FileSource {
+    enum Privacy {
+        UNSET = 0;
+        PUBLIC = 1;
+        PRIVATE = 2;
+    }
     oneof data {
         google.protobuf.StringValue url = 1; // Absolute URL
-        google.protobuf.StringValue custom = 2; // ID that will be resolved to a URL by a resolver function
+        google.protobuf.StringValue custom = 2 [deprecated = true]; // Custom id (string)
```
Comment (mharsat) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:83]: Nice usage of the deprecated option!
This requires a change to our `generateSchema` script. It currently removes any field option, we should change the regex to remove only wix options/validations

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto [hunk 5]
```diff
@@ -73,10 +73,18 @@ message PluginContainerData {
...
+        UNSET = 0;
+        PUBLIC = 1;
+        PRIVATE = 2;
+    }
     oneof data {
         google.protobuf.StringValue url = 1; // Absolute URL
-        google.protobuf.StringValue custom = 2; // ID that will be resolved to a URL by a resolver function
+        google.protobuf.StringValue custom = 2 [deprecated = true]; // Custom id (string) 
+        google.protobuf.StringValue id = 3; // ID that will be resolved to a URL by a resolver function
+        google.protobuf.StringValue json = 4; // Custom string for consumers with an ex-schematic source object
```
Comment (mharsat) [inline packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:85]:
```suggestion
        google.protobuf.StringValue json = 4; // For custom string values or string objects
```

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto (hunk 4 above)
Comment (mharsat) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto:83]:
```suggestion
        google.protobuf.StringValue custom = 2 [deprecated = true]; // Custom ID
```

**Review by mharsat — CHANGES_REQUESTED** [review]

**Review by mharsat — APPROVED** [review]
LGTM 🎉

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto (hunk 4 above)
Comment (shaulgo) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto]: do we need to deprecate it? did anyone use this plugin?

File: website/docs/plugins_api/FileUploadPlugin.mdx [hunk 6]
```diff
@@ -62,6 +62,7 @@ data?: {
   type: string;
   url?: string;
   id?: string;
+  privacy?: 'public' | 'private';
```
Comment (shaulgo) [inline website/docs/plugins_api/FileUploadPlugin.mdx:65]: maybe private = true/false

File: packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto [hunk 7]
```diff
@@ -79,10 +79,18 @@ message PluginContainerData {
...
+        UNSET = 0;
+        PUBLIC = 1;
+        PRIVATE = 2;
+    }
     oneof data {
         google.protobuf.StringValue url = 1; // Absolute URL
-        google.protobuf.StringValue custom = 2; // ID that will be resolved to a URL by a resolver function
+        google.protobuf.StringValue custom = 2 [deprecated = true]; // Custom ID
+        google.protobuf.StringValue id = 3; // ID that will be resolved to a URL by a resolver function
+        google.protobuf.StringValue json = 4; // For custom string values or string objects
```
Comment (shaulgo) [inline packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto:91]: is json used?

File: packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto [hunk 8]
```diff
@@ -79,10 +79,18 @@ message PluginContainerData {
 }
 
 message FileSource {
+    enum Privacy {
```
Comment (shaulgo) [inline packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto:82]: why not private: bool?

Comment (Barackos) [reply packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto:82]: I'll update Schema 👍🏻.

However we should keep `privacy: 'public' | 'private'` in draft, due to its existence in current contents. I'll address it in the converters

File: packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto (hunk 7 above)
Comment (Barackos) [reply packages/ricos-schema/web/src/generated/proto/wix/rich_content/v1/common.proto:91]: Not in the converters. We wanted to allow consumers to add customizations of their choice, but only for the new schema.

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto (hunk 4 above)
Comment (Barackos) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto]: Yup, e.g. WixData

**Review by lxgreen — APPROVED** [review]

File: packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto (hunk 4 above)
Comment (shaulgo) [reply packages/ricos-schema/web/src/main/proto/wix/rich_content/v1/common.proto]: I mean if we can find out if there is a site that is using it?

**Pull request merged**